lib/services/MetricService.ts

```diff
@@ -0,0 +1,206 @@
+            labelNames: ['protocol'], 
+            registers: [this.registries.core] 
+          }),
+        },
+        realtime: {
+          rooms: new Gauge({
+            name: `${config.metrics.core.prefix}realtime_rooms`, 
+            help: 'Number of rooms', 
+            registers: [this.registries.core] 
+          }),
+          subscriptions: new Gauge({
+            name: `${config.metrics.core.prefix}realtime_subscriptions`,
+            help: 'Number of subscriptions', 
+            registers: [this.registries.core] 
+          }),
+        },
+      },
+    };
+
+    if (config.metrics.core.monitorRequestDuration) {
+      // Need to be registered in a separate registry to avoid
+      // the core registry resetting in updateCoreMetrics
+      this.registries.requestDuration = new Registry();
+      this.metrics.requestDuration = new Histogram({
+        name: `${config.metrics.core.prefix}api_request_duration_ms`,
+        help: 'Duration of Kuzzle requests in ms',
+        labelNames: ['action', 'controller', 'protocol', 'status'],
+        registers: [this.registries.requestDuration],
+        buckets: [0.10, 5, 15, 50, 100, 200, 300, 400, 500]
+      });
+    }
+
+    if (config.metrics.default.enabled) {
+      this.registries.default = new Registry();
+      collectDefaultMetrics({
+        register: this.registries.default,
+        prefix: config.metrics.default.prefix,
+        eventLoopMonitoringPrecision: config.metrics.default.eventLoopMonitoringPrecision,
+        gcDurationBuckets: config.metrics.default.gcDurationBuckets,
+      });
+    }
+  }
+
+  /**
+   * Update the Prometheus coreMetrics with from the server:metrics JSON response
+   * @param {JSONObject} jsonMetrics - The server:metrics JSON response
+   */
+  public updateCoreMetrics(jsonMetrics: JSONObject): void {
+    // Past metrics are oudated, so we need to reset them
+    this.registries.core.resetMetrics();
+
+    for (const component of Object.keys(jsonMetrics)) {
+      for (const metric of Object.keys(jsonMetrics[component])) {
+        if (typeof this.metrics.core[component][metric] === 'undefined') {
+          continue;
+        }
+
+        if (typeof jsonMetrics[component][metric] === 'number') {
+          this.metrics.core[component][metric].set(jsonMetrics[component][metric]);
+        }
+
+        // Only for network.connections metric since we label it using protocol name
+        if (typeof jsonMetrics[component][metric] === 'object') {
+          for (const protocol of Object.keys(jsonMetrics[component][metric])) {
+            this.metrics.core[component][metric].set({ protocol }, jsonMetrics[component][metric][protocol]);
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Merge all the Prometheus registries into one and returns metrics as Prometheus text format
+   * @returns {string} All the regitries metrics formatted as a Prometheus text
+   */
+  public async getMetrics(): Promise<string> {
```

Comment:
Nitpicking: coding standard
```suggestion
  public async getMetrics (): Promise<string> {
```